src/csharp/Microsoft.Spark/Sql/RelationalGroupedDataset.cs

```diff
@@ -85,6 +85,40 @@ internal RelationalGroupedDataset(JvmObjectReference jvmObject, DataFrame dataFr
         public DataFrame Sum(params string[] colNames) =>
             new DataFrame((JvmObjectReference)_jvmObject.Invoke("sum", (object)colNames));
 
+        /// <summary>
+        /// Pivots a column of the current DataFrame and performs the specified aggregation.
+        /// </summary>
+        /// <param name="pivotColumn">Name of the column to pivot</param>
+        /// <returns>New RelationalGroupedDataset object with pivot applied</returns>
+        public RelationalGroupedDataset Pivot(string pivotColumn) =>
+            new RelationalGroupedDataset((JvmObjectReference)_jvmObject.Invoke("pivot", pivotColumn), _dataFrame);
+
+        /// <summary>
+        /// Pivots a column of the current DataFrame and performs the specified aggregation.
+        /// </summary>
+        /// <param name="pivotColumn">Name of the column to pivot</param>
+        /// <param name="values">List of values that will be translated to columns in the output DataFrame</param>
+        /// <returns>New RelationalGroupedDataset object with pivot applied</returns>
+        public RelationalGroupedDataset Pivot(string pivotColumn, params object[] values) =>
+            new RelationalGroupedDataset((JvmObjectReference)_jvmObject.Invoke("pivot", pivotColumn, values), _dataFrame);
+
+        /// <summary>
+        /// Pivots a column of the current DataFrame and performs the specified aggregation.
+        /// </summary>
+        /// <param name="pivotColumn">The column to pivot</param>
+        /// <returns>New RelationalGroupedDataset object with pivot applied</returns>
+        public RelationalGroupedDataset Pivot(Column pivotColumn) =>
+            new RelationalGroupedDataset((JvmObjectReference)_jvmObject.Invoke("pivot", pivotColumn), _dataFrame);
+
+        /// <summary>
+        /// Pivots a column of the current DataFrame and performs the specified aggregation.
+        /// </summary>
+        /// <param name="pivotColumn">The column to pivot</param>
+        /// <param name="values">List of values that will be translated to columns in the output DataFrame</param>
+        /// <returns>New RelationalGroupedDataset object with pivot applied</returns>
+        public RelationalGroupedDataset Pivot(Column pivotColumn, params object[] values) =>
+            new RelationalGroupedDataset((JvmObjectReference)_jvmObject.Invoke("pivot", pivotColumn, values), _dataFrame);
```

Comment:
@imback82 it seems like passing object arrays is not supported by the interop layer, according to the failing tests. What would you suggest I should do here instead? How about `Pivot<T>(Column pivotColumn, params T[] values)`, that might work?

Reply:
@moredatapls Can we get this PR in without exposing the `pivot` APIs requiring `values` for now? We can discuss the solution for that separately, if you're interested in contributing, it involves adding Serialization ([here](https://github.com/dotnet/spark/blob/master/src/csharp/Microsoft.Spark/Interop/Ipc/PayloadHelper.cs)) and Deserialization ([here](https://github.com/dotnet/spark/blob/master/src/scala/microsoft-spark-2.4.x/src/main/scala/org/apache/spark/api/dotnet/SerDe.scala)) of an array of `objects`.

Reply:
I agree, let's focus on the APIs with the single parameters for now and tackle the others in a separate PR.